filament/src/components/RenderableManager.cpp

```diff
@@ -408,10 +409,12 @@ void FRenderableManager::create(
                         backend::BufferUsage::DYNAMIC),
                 .count = targetCount };
 
-            for (size_t i = 0, c = builder->mEntries.size(); i < c; ++i) {
+            MorphTargets* morphTargets = new MorphTargets[entryCount];
+            for (size_t i = 0; i < entryCount; ++i) {
                 const auto& morphing = builder->mEntries[i].morphing;
-                rp[i].set(upcast(morphing.buffer));
+                morphTargets[i] = { upcast(morphing.buffer), morphing.offset, morphing.count };
             }
+            mManager[ci].morphTargets = { morphTargets, size_type(entryCount) };
```

Comment:
Why not store `MorphTargets` directly into `mManager[ci].morphTargets` instead of allocating an array on the heap?
There is no advantage because the array is always allocated, even if there is no morphing.

Reply:
Because each primitive needs a different morph target buffer. If we store `MorphTargets` directly into `mManager[ci].morphTargets` then all primitives have a same morph target buffer.

This array will be allocated if skinning or morphing is enabled.

Reply:
The goal of this PR is not saving memory. The goal is that changing code to make sense more.

Having morph targets in primitive is a concept of glTF. Filament doesn't have to follow this concept. Actually the resposiblity of `FRenderPrimitive` is to define vertex input stream because `FRenderPrimitive` map to `HwRenderPrimitive` in Filament. If we store `MorphTargetBuffer` into `FRenderPrimitive` then it exceeds the resposibility of `FRenderPrimitive`. Becase `MorphTargetBuffer` doesn't have any information how to set up vertex input stream. So handling morphing like skinning will be better.